File: examples/undocumented/python_modular/kernel_io_modular.py
#!/usr/bin/env python
traindat = '../data/fm_train_real.dat'
testdat = '../data/fm_test_real.dat'

parameter_list=[[traindat,testdat,1.9],[traindat,testdat,1.7]]

def kernel_io_modular (train_fname=traindat,test_fname=testdat,width=1.9):
	from modshogun import RealFeatures, GaussianKernel, CSVFile
	
	feats_train=RealFeatures(CSVFile(train_fname))
	feats_test=RealFeatures(CSVFile(test_fname))

	kernel=GaussianKernel(feats_train, feats_train, width)
	km_train=kernel.get_kernel_matrix()
	f=CSVFile("gaussian_train.csv","w")
	kernel.save(f)
	del f

	kernel.init(feats_train, feats_test)
	km_test=kernel.get_kernel_matrix()
	f=CSVFile("gaussian_test.csv","w")
	kernel.save(f)
	del f

	#clean up
	import os
	os.unlink("gaussian_test.csv")
	os.unlink("gaussian_train.csv")
	
	return km_train, km_test, kernel

if __name__=='__main__':
	print('Gaussian')
	kernel_io_modular(*parameter_list[0])

File: examples/undocumented/csharp_modular/kernel_io_modular.cs
using System;

public class kernel_io_modular {
	public static void Main() {
		modshogun.init_shogun_with_defaults();
		double width = 1.2;

		double[,] traindata_real = Load.load_numbers("../data/fm_train_real.dat");
		double[,] testdata_real = Load.load_numbers("../data/fm_test_real.dat");

		RealFeatures feats_train = new RealFeatures(traindata_real);
		RealFeatures feats_test = new RealFeatures(testdata_real);

		GaussianKernel kernel = new GaussianKernel(feats_train, feats_test, width);
		double[,] km_train = kernel.get_kernel_matrix();
		CSVFile f=new CSVFile("gaussian_train.ascii",'w');
		kernel.save(f);

		kernel.init(feats_train, feats_test);
		double[,] km_test = kernel.get_kernel_matrix();
		CSVFile f_test=new CSVFile("gaussian_train.ascii",'w');
		kernel.save(f_test);

		//  Parse and Display km_train
		Console.Write("km_train:\n");
		int numRows = km_train.GetLength(0);
		int numCols = km_train.GetLength(1);

		for(int i = 0; i < numRows; i++){
			for(int j = 0; j < numCols; j++){
				Console.Write(km_train[i,j] +" ");
			}
			Console.Write("\n");
		}

		//  Parse and Display km_test
		Console.Write("\nkm_test:\n");
		numRows = km_test.GetLength(0);
		numCols = km_test.GetLength(1);

		for(int i = 0; i < numRows; i++){
			for(int j = 0; j < numCols; j++){
				Console.Write(km_test[i,j] +" ");
			}
			Console.Write("\n");
		}

	}
}
